src/Controller/PayPalButtonsController.php

```diff
@@ -129,4 +144,39 @@ public function renderPaymentPageButtonsAction(Request $request): Response
             return new Response('');
         }
     }
+
+    public function renderPayPalPaymentAction(Request $request): Response
+    {
+        $orderId = $request->attributes->getInt('orderId');
+        /** @var OrderInterface $order */
+        $order = $this->orderRepository->find($orderId);
+        /** @var PaymentInterface $payment */
+        $payment = $order->getLastPayment();
+        /** @var PaymentMethodInterface $paymentMethod */
+        $paymentMethod = $payment->getMethod();
+        /** @var GatewayConfigInterface $gatewayConfig */
+        $gatewayConfig = $paymentMethod->getGatewayConfig();
```

Comment:
There are so many places where we get data from the gateway config :/ We should definitely extract some service for that (but, of course, not in this PR 🖖)